Hex Calculator/HexCalc.java

```diff
@@ -0,0 +1,446 @@
+package main;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+/**
+ * HexCalc class is used to perform different operations on a hexadecimal number
+ * @author Harshit
+ *
+ */
+public class HexCalc implements GeneralisedCalculator
+{
+    final private static char[] check={'0','1','2','3','4','5','6','7','8','9','A','B','C','D','E','F','a','b','c','d','e','f'};
+    
+    /**
+     * Addition of two Hexadecimal numbers
+     * @param firstHexaNumber denotes first Hexadecimal number
+     * @param secondHexaNumber denotes second Hexadecimal number
+     * @return Added Hexadecimal numbers in the form of a string
+     */
+    @Override
+    public String addNumbers(String firstHexaNumber,String secondHexaNumber)
+    {
+        int firstDecimalNumber=Integer.parseInt(this.convertToDecimal(firstHexaNumber)),secondDecimalNumber=Integer.parseInt(this.convertToDecimal(secondHexaNumber));
+        int addDecimalNumber=(firstDecimalNumber+secondDecimalNumber);
+        String addHexadecimalnumbers=Integer.toString(addDecimalNumber);
+       
+        addHexadecimalnumbers=convertToHexaDecimal(addHexadecimalnumbers);
+        return (addHexadecimalnumbers);

```

Comment:
Instead of creating new variable here, try to achieve the same result with less no of lines of code and less variables